auction.sympl

```diff
@@ -33,95 +40,273 @@ schema Auction:
 
-# Create an auction.
-@clientside
-def create_auction(product_description: str
-                  , initial_price: int) -> None:
-    cvm.job_start()
-    with PostTxArgs(PUBLIC):
-        _create_auction(product_description, initial_price)
 
-# Executable to create an auction.
-# This executable tracks the creator of the auction for later use.
-@executable
-def _create_auction(product_description: str
-                   , initial_price: int) -> None:
-    id = cvm.generate_id('AUC')
-    creator : str = cvm.tx.key_alias
-    auction = cvm.storage.get(PUBLIC, AuctionStatic, id)
-    if isinstance(auction, Auction):
-        cvm.error(f"Auction already exist id={id}")
-    new_auction = Auction(id = id
-                         , product_description = product_description
-                         , initial_price = initial_price
-                         , last_bid = None
-                         , creator = creator
-                         , closed = False)
-    cvm.storage.put(id, new_auction)
-    cvm.job_complete(std.json(new_auction))
+###############################
+# Bid API and Implementations #
+###############################
 
-# Place a bid on a product (idenfitied by the auction id)
+# Places a bid on a product, idenfitied by the auction ID.
 @clientside
-def bid(product_id: Identifier, amount: int) -> None:
-    cvm.job_start()
-    with PostTxArgs(PUBLIC):
+def bid(product_id: Identifier, amount: int, channel: ChannelName=PUBLIC) -> None:
+    with PostTxArgs(channel):
         _bid(product_id, amount)
 
-# Executable to place a bid on a product (idenfitied by the auction id)
 @executable
 def _bid(product_id: Identifier, amount: int) -> None:
     bidder: str = cvm.tx.key_alias
-    auction = cvm.storage.get(PUBLIC, AuctionStatic, product_id)
+    channel : ChannelName = cvm.tx.write_channel
+    auction = cvm.storage.get(channel, AuctionStatic, product_id)
     if isinstance(auction, None):
-        cvm.error(f"Unknown product id={product_id}")
+        cvm.error(f"Unknown product {product_id}")
     if auction.closed:
-        cvm.error(f"Closed auction id={product_id}")
+        cvm.error(f"Closed auction {product_id}")
     last_bid = auction.last_bid
     if isinstance(last_bid, None):
         if amount < auction.initial_price:
-            cvm.error("The bid amount must be superior to initial price id={product_id}")
+            cvm.error(f"The bid amount must be larger than initial price ${auction.initial_price}.")
     else:
         if amount <= last_bid.amount:
-            cvm.error("The bid amount must be superior to previous bid amount id={product_id}")
+            cvm.error(f"The bid amount must be larger than previous bid amount ${last_bid.amount}.")
     bid = Bid(bidder = bidder, amount = amount)
     auction.last_bid = bid
     cvm.storage.put(product_id, auction)
-    cvm.job_complete(std.json(auction))
+    cvm.create_event('Bid Placed', std.json(auction))
```

Comment:
same comment as above. The auction will be revealed to the whole node where a KA is allowed to see it.